src/libs/EmojiTrie.js

```diff
@@ -0,0 +1,31 @@
+import emojis from '../../assets/emojis';
+import Trie from './Trie';
+
+// Create a Trie object
+const emojisTrie = new Trie();
+
+// Inserting all emojis into the Trie object
+for (let i = 0; i < emojis.length; i++) {
+    if (emojis[i].name) {
+        const node = emojisTrie.isWord(emojis[i].name);
+        if (!node) {
+            emojisTrie.add(emojis[i].name, {code: emojis[i].code, suggestions: []});
+        } else {
+            emojisTrie.update(emojis[i].name, {code: emojis[i].code, suggestions: node.getMetaData().suggestions});
```

Comment:
I'm confused by this ... what is happening here? Only the emoji code is being changed, but not the metadata?

Reply:
Sometimes the node is exist in the Trie and we only need to update the node suggestions metadata.

i.e  when adding these two emojis into the Trie
```javascript
         code: '😀',
        'keywords': [
          'face',
          'grin'
        ],
        name: 'grinning'
      }

      code: '😉',
      'keywords': [
        'face',
        'wink'
      ],
      name: 'wink'
```
For the first emoji we will add these 3 nodes :
```javascript
{name: 'grinning', metadata: {code: '😀', suggestions: []}}
{name: 'face', metadata: {suggestions: [{code: '😀', name: 'grinning'}]}}
{name: 'grin', metadata: {suggestions: [{code: '😀', name: 'grinning'}]}}
```
But for the second emoji, we can’t add this node :
```javascript
{name: 'face', metadata: {suggestions: [{code: '😀', name: 'grinning’}]}}
```
Because we added the `face` node before, so instead we will update it and it will look like this: 
```javascript
{name: 'face', metadata: {suggestions: [{code: '😀', name: 'grinning’}, {code: ‘😉’, name: ‘wink’}]}}
```